public/js/roles/roles-list.controller.js

```diff
@@ -64,5 +65,30 @@
                 vm.getRoles(vm.page);
             });
         }
+
+        function deleteRole (roleId) {
+            api.getUsers().then(function (res) {
+                const usersHasRole = res.data.filter(e => e.roles.includes(roleId));
+                const targetRole = vm.items.filter(e => { return e._id === roleId; })[0];
+
+                const modal = $uibModal.open({
+                    component: 'appDeleteModal',
+                    resolve: {
+                        title: () => gettextCatalog.getString('Remove role {{name}} ? - {{usersCount}} user(s) affected', { name: targetRole.name, usersCount: usersHasRole.length }),
+                        delete: () => function () {
+                            if (usersHasRole.length > 0) {
+                                usersHasRole.forEach(u => api.deleteUserRole(u._id, roleId));
+                            }
+                            return api.deleteRole(roleId);
+                        },
+                    },
+                });
+                modal.result.then(function () {
+                    $rootScope.$broadcast('counts-changes');
+                    notify.success(gettextCatalog.getString('Role deleted successfully'));
+                    vm.getRoles(1);
+                });
```

Comment:
You should handle promise rejection (when the user close the modal by clicking outside the modal for instance), otherwise you will get an error message in the console (`Possibly unhandled rejection`)

Reply:
add catch to deal with unhandled rejection